Automattic/woocommerce-payments
# Payouts: Show bank reference key on payout details page

Fixes #9878 

#### Changes proposed in this Pull Request

* Show bank reference key of a payout on its details page, when it is available.  The PR tentatively adds it to the top header. Depending on feedback from stakeholders we can have this shifted / modified. Screenshot of how it appears:

<img src="https://github.com/user-attachments/assets/ddf47ea7-531c-43b5-934a-0c04472e70c3" width="600px" />

Note that in test payouts, the bank reference key is of 41 characters length. In real payouts, we see a variety of lengths, sometimes much smaller ( 12 characters ).

* There is a copy button added too for easy copying by merchant.

* The bank reference key shows as `N/A` if it is not available from Stripe.

<img src="https://github.com/user-attachments/assets/5016b6f8-68e7-4f1c-aa1b-4ece50c4f1fa" width="600px" />

* The PR also adds a few CSS fixes for mobile view. This is needed to change the alignment of the summary items as column, and adjust borders + text alignment for smaller screens < 660px . 

<img src="https://github.com/user-attachments/assets/89d4ead4-a648-47fd-a91e-1d45aa0bb9d8" width="500px" />

#### Testing instructions

* Use a test account that preferably has payouts from before December 2, 2024, to test with payouts without the bank reference key. You may use `acct_1OuYn1QpUShfetUA` , if you like.
* Checkout this branch on your client and run `npm start` to build assets.
* Browse to `Payments > Payouts` 
* Click on any Payout where you see a Bank reference key 
* You should see the same key on the top, on the details page when it opens. Similar to the first screenshot above.
* Click the `Copy` button beside the Bank reference key, and see that it copies the key to clipboard. 
* Back on the payouts list page, click on any payout without a bank reference key
* You should not see the bank reference key mentioned on the header of the details page. Similar to the second screenshot above.
* Test on mobile screen too 


-------------------


**Post merge**
